packages/botonic-api/src/websocket/local-websocket-server.ts

```diff
@@ -1,29 +1,43 @@
 import chalk from 'chalk'
+import express from 'express'
+import expressWs from 'express-ws'
 import { v4 } from 'uuid'
-import { Server } from 'ws'
 
-export const localWebSocketServer = ({
-  onConnect,
-  onMessage,
-  onDisconnect,
-}) => {
-  const wss = new Server({ port: parseInt(process.env.PORT as string) })
-  console.log(
-    `${chalk.bold('Botonic WebSocket API')} listening on port ${chalk.bold(
-      process.env.PORT
-    )}`
-  )
-  wss.on('connection', function connection(ws: any) {
+export const localWebSocketServer = ({ onConnect, onAuth, onDisconnect }) => {
+  const connections = {}
+  const wsApp = expressWs(express()).app
+  wsApp.use(express.json())
+  wsApp.use(express.urlencoded({ extended: true }))
+
+  wsApp.post('/send', (req, res) => {
+    const { botEvent, websocketId } = req.body
+    connections[websocketId].send(JSON.stringify(botEvent))
+    res.status(200)
+    res.end() // TODO: is res.end necessary?
+  })
+
+  wsApp.ws('/', function (ws: any, req) {
     ws.id = v4()
+    connections[ws.id] = ws
     onConnect(ws.id)
-    ws.on('message', async function incoming(data) {
-      onMessage({
+    ws.on('message', function (data) {
+      onAuth({
         websocketId: ws.id,
         data,
         send: message => ws.send(JSON.stringify(message)),
       })
     })
```

Comment:
thought for future consideration: we're now binding ALL websocket incoming messages to the `onAuth` function. It would be cool to allow the developer (or plugins) to extend the websocket functionality in other ways, this means that we should be able to route incoming messages to other functions.
One example of this could be a "co-browsing" functionality, where the webchat would send mouse position continuously through the websocket.